src/agentlab/agents/step_agent/agent.py

```diff
@@ -0,0 +1,57 @@
+from typing import List
+
+
+class Agent:
+    def __init__(
+        self,
+        max_actions,
+        verbose=0,
+        logging=False,
+        previous_actions: List[str] = None,
+        previous_reasons: List[str] = None,
+        previous_responses: List[str] = None,
+    ):
+        self.previous_actions = [] if previous_actions is None else previous_actions 
+        self.previous_reasons = [] if previous_reasons is None else previous_reasons
+        self.previous_responses = [] if previous_responses is None else previous_responses
+        self.max_actions = max_actions
+        self.verbose = verbose
+        self.logging = logging
+        self.trajectory = []
+        self.data_to_log = {}
+
+    def reset(self):
+        self.previous_actions = []
+        self.previous_reasons = []
+        self.previous_responses = []
+        self.trajectory = []
+        self.data_to_log = {}
+
+    def get_trajectory(self) -> list[str]:
+        return self.trajectory
+    
+    def update_history(self, action: str, reason: str):
+        if action:
+            self.previous_actions += [action]
+        if reason:
+            self.previous_reasons += [reason]    
+
+    def predict_action(self, objective: str, observation: str, url: str=None):
+        pass
+
+    def receive_response(self, response: str):
+        self.previous_responses += [response]
+
+    def log_step(self, objective: str, url: str, observation: str, action: str, reason: str, status: str):
```

Comment:
The `log_step` isn't actually being invoked by the `Agent` class. The original `Agent` class invoked `log_step` in the `act()` function: https://github.com/asappresearch/webagents-step/blob/a24086f948daf14dd1a034e7f8e7e77953aad41f/src/webagents_step/agents/agent.py#L45

Since this implementation does away with the `act` function, we may want to call `log_step` wherever the agent is made to predict actions, most likely in the `BrowserGymStepAgent` class